monai/application/datasets.py

```diff
@@ -0,0 +1,114 @@
+            if expected file already exists, skip downloading even set it to True.
+            user can manually copy `MedNIST.tar.gz` file or `MedNIST` folder to root directory.
+        extract: whether to extract the MedNIST.tar.gz file under root directory, default is False.
+        seed: random seed to randomly split training, validation and test datasets, defaut is 0.
+        cache_num: number of items to be cached. Default is `sys.maxsize`.
+            will take the minimum of (cache_num, data_length x cache_rate, data_length).
+        cache_rate: percentage of cached data in total, default is 1.0 (cache all).
+            will take the minimum of (cache_num, data_length x cache_rate, data_length).
+        num_workers: the number of worker threads to use.
+            If 0 a single thread will be used. Default is 0.
+
+    """
+
+    resource = "https://www.dropbox.com/s/5wwskxctvcxiuea/MedNIST.tar.gz?dl=1"
+    md5 = "0bc7306e7427e00ad1c5526a6677552d"
+
+    def __init__(
+        self,
+        root: str,
+        section: str,
+        transform: Callable[..., Any],
+        download: bool = False,
+        extract: bool = False,
+        seed: int = 0,
+        cache_num: int = sys.maxsize,
+        cache_rate: float = 1.0,
+        num_workers: int = 0,
+    ):
+        if not os.path.isdir(root):
+            raise ValueError("root must be a directory.")
+        self.section = section
+        self.set_random_state(seed=seed)
+        tarfile_name = os.path.join(root, "MedNIST.tar.gz")
+        dataset_dir = os.path.join(root, "MedNIST")
+        if download:
+            download_url(self.resource, tarfile_name, self.md5)
+        if extract:
+            extractall(tarfile_name, root)
+        if not os.path.exists(dataset_dir):
+            raise RuntimeError("can not find dataset directory, please use download=True to download it.")
+        data = self._generate_data_list(dataset_dir)
+        super().__init__(data, transform, cache_num=cache_num, cache_rate=cache_rate, num_workers=num_workers)
+
+    def randomize(self):
+        self.rann = self.R.random()
+
+    def _generate_data_list(self, dataset_dir):
+        class_names = sorted([x for x in os.listdir(dataset_dir) if os.path.isdir(os.path.join(dataset_dir, x))])
+        num_class = len(class_names)
+        image_files = [
+            [
+                os.path.join(dataset_dir, class_names[i], x)
+                for x in os.listdir(os.path.join(dataset_dir, class_names[i]))
+            ]
+            for i in range(num_class)
+        ]
+        num_each = [len(image_files[i]) for i in range(num_class)]
+        image_files_list = []
+        image_class = []
+        for i in range(num_class):
+            image_files_list.extend(image_files[i])
+            image_class.extend([i] * num_each[i])
+        num_total = len(image_class)
+
+        val_frac = 0.1
+        test_frac = 0.1
```

Comment:
perhaps val_frac and test_frac could be specified by the users as well?

we'll need another iteration to have APIs for training/validation/test splitting.

Reply:
I think this MedNISTDataset is just for quick start tutorial, so didn't add it to args.
Anyway, I added in the latest commit now.
And actually, I plan to add logic to split dataset for MSD datasets later.